src/libs/antares/enums/class_name.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2007-2025, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#include <antares/enums/class_name.h>
+
+namespace stdcxx
+{
+
+std::string simpleClassName(const char* className)
+{
+    const std::string& strClassName = className;
+    std::size_t index = strClassName.find_last_of("::");
+
+    return (index == std::string::npos) ? strClassName
+                                        : strClassName.substr(index + 1, strClassName.size());
+}
```

Comment:
Issues
1. lifetime of temporary `std::string` isn't garuanteed beyond l. 29
2. `find_last_of` doesn't behave as expected, in some cases it fails to find the appropriate substring. For example "A:::B" -> ":B". On the other hand, `rfind("::")` searches for the last occurrence of the exact substring "::", not individual colons.
```suggestion
std::string simpleClassName(const char* className)
{
    const std::string strClassName(className);
    std::size_t index = strClassName.rfind("::");

    return (index == std::string::npos) ? strClassName
                                        : strClassName.substr(index + 2); // +2 because "::" has length 2
}
}
```